lute/std/libs/json.luau

```diff
@@ -278,6 +278,17 @@ local function deserializeNumber(state: DeserializerState)
 	return num
 end
 
+local function decodeSurrogatePair(high, low): string?
+	local highVal = tonumber(high, 16)
+	local lowVal = tonumber(low, 16)
+	if not highVal or not lowVal then
+		return nil -- Invalid
+	end
+	-- Calculate the actual Unicode codepoint
+	local codepoint = 0x10000 + ((highVal - 0xD800) * 0x400) + (lowVal - 0xDC00)
```

Comment:
These are very magic numbers to me - is there something I can read to understand these better? Would you be able to add a more detailed comment here explaining why this works (or is this just a standard unicode transform).